lib/deliver/submit_for_review.rb

```diff
@@ -29,10 +29,19 @@ def submit!(options)
     end
 
     def select_build(options)
-      Helper.log.info "Selecting the latest build..."
       app = options[:app]
       v = app.edit_version
-      build = wait_for_build(app)
+
+      if options[:build_number]
+        Helper.log.info "Selecting existing build-number: #{options[:build_number]}"
+        build = v.candidate_builds.find(build_version: options[:build_number]).first
```

Comment:
has this been tested for other values than 'latest' ? I don't think this works. (find takes a block not parameters)